ark/system/genesis/genesis_backend.py

```diff
@@ -0,0 +1,349 @@
+"""@file pybullet_backend.py
+@brief Backend implementation for running simulations in PyBullet.
+"""
+
+from __future__ import annotations
+
+import importlib.util
+import sys
+from pathlib import Path
+from typing import Any
+
+import cv2
+import genesis as gs
+import numpy as np
+
+from ark.tools.log import log
+from ark.system.simulation.simulator_backend import SimulatorBackend
+
+from ark.system.genesis.genesis_multibody import GenesisMultiBody
+# from ark.system.genesis.genesis_camera_driver import GenesisCameraDriver
+from arktypes import *
+
+
+def import_class_from_directory(path: Path) -> tuple[type[Any], Any | None]:
```

Comment:
importing functions/classes from a path should be in a utils script, the part specific to genesis can be part of a function here